ballista/core/src/serde/mod.rs

```diff
@@ -114,6 +118,100 @@ impl<T: 'static + AsLogicalPlan, U: 'static + AsExecutionPlan> BallistaCodec<T,
     }
 }
 
+#[derive(Debug, Clone)]
+pub struct BallistaLogicalExtensionCodec {
```

Comment:
had to create a `BallistaLogicalExtensionCodec` which combines all the file format codecs, not sure if there is a better way atm